nucliadb_node/src/services/reader.rs

```diff
@@ -298,26 +299,20 @@ impl ShardReaderService {
     /// defines the limit of words a query can have.
     #[tracing::instrument(skip_all)]
     fn split_suggest_query(query: String, max_group: usize) -> Vec<String> {
-        let words = query.split(' ');
-        let mut i = 0;
-        let mut prefixes = vec![];
-        let mut prefix = String::new();
-
-        for word in words.rev() {
-            if prefix.is_empty() {
-                prefix = word.to_string();
-            } else {
-                prefix = format!("{word} {prefix}");
-            }
-            prefixes.push(prefix.clone());
-
-            i += 1;
-            if i == max_group {
-                break;
+        // Paying the price of allocating the vector to not have to
+        // prepend to the partial strings.
+        let relevant_words: Vec<_> = query.split(' ').rev().take(max_group).collect();
+        let mut prefixes = vec![String::new(); max_group];
+        for (index, word) in relevant_words.into_iter().rev().enumerate() {
+            // The inner loop is upper-bounded by max_group
+            for prefix in prefixes.iter_mut().take(index + 1) {
+                if !prefix.is_empty() {
+                    prefix.push(' ');
+                }
+                prefix.push_str(word);
```

Comment:
I don't get why this two loops are more efficient than one.

With something like:
```rust
for word in words.rev().take(max_group)
```
you can remove the need for the counter and iterate only one time per word.